### PR TITLE
Create user controller

### PR DESCRIPTION
**Pull request by alexdavcudriz**
## Summary
- User controller was created.
- User service was created.
- User not found exception was created.
- Book service was created too.
## Trello Card
https://trello.com/c/2rXCBxAv/21-crear-user-controller-con-las-operaciones-crud-de-user

### CODE REVIEW

File: src/main/java/wolox/training/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
```
Comment (jdhiguita) [inline src/main/java/wolox/training/controllers/UserController.java]: Although the card does not mention it, a good practice is to add service classes in the controllers instead repositories and the services have them implement the repositories

File: src/main/java/wolox/training/controllers/UserController.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    /**
+     * User updating method
+     *
+     * @param user: Request body ({@link User})
+     * @param id:   User identifier (Long)
+     * @return {@link User}
+     */
+    @PutMapping("/{id}")
+    public User update(@RequestBody User user, @PathVariable Long id) {
+        User userFound = userRepository.findById(id).orElseThrow(UserNotFoundException::new);
```
Comment (jdhiguita) [inline src/main/java/wolox/training/controllers/UserController.java]: This implemantion must be in the service

File: src/main/java/wolox/training/controllers/UserController.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+     * Method that adds a book
+     *
+     * @param idUser: User identifier (Long)
+     * @param idBook: Book identifier (Long)
+     * @return {@link User}
+     */
+    @PutMapping("/{idUser}/books/{idBook}")
+    public User addBook(@PathVariable Long idUser, @PathVariable Long idBook) {
+
+        User userFound = userRepository.findById(idUser)
```
Comment (jdhiguita) [inline src/main/java/wolox/training/controllers/UserController.java]: The same comment above

**Review by jdhiguita — COMMENTED** [review]
Good job! Some comments

**Review by jdhiguita — APPROVED** [review]

File: src/main/java/wolox/training/controllers/UserController.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
+package wolox.training.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
```
Comment (facucompe) [inline src/main/java/wolox/training/controllers/UserController.java:5]: Remember removing wildcards

**Review by facucompe — APPROVED** [review]

**Pull request merged**